java/java.lsp.server/vscode/src/test/suite/extension.test.ts

```diff
@@ -97,6 +99,88 @@ class Main {
     test("Compile workspace6", async() => demo(6));
 //    test("Compile workspace7", async() => demo(7));
 //    test("Compile workspace8", async() => demo(8));
+
+    /**
+     * Checks that maven-managed process can be started, and forcefully terminated by vscode
+     * although it does not run in debugging mode.
+     */
+    async function mavenTerminateWithoutDebugger() {
+        let folder: string = assertWorkspace();
+
+        await fs.promises.writeFile(path.join(folder, 'pom.xml'), `
+    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.netbeans.demo.vscode.t1</groupId>
+    <artifactId>basicapp</artifactId>
+    <version>1.0</version>
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+    </properties>
+    </project>
+        `);
+
+        let pkg = path.join(folder, 'src', 'main', 'java', 'pkg');
+        let mainJava = path.join(pkg, 'Main.java');
+
+        await fs.promises.mkdir(pkg, { recursive: true });
+
+        await fs.promises.writeFile(mainJava, `
+    package pkg;
+    class Main {
+    public static void main(String... args) throws Exception {
+        System.out.println("Endless wait...");
+        while (true) {
+            Thread.sleep(1000);
+        }
+    }
+    }
+        `);
+        vscode.workspace.saveAll();
+        let u : Uri = vscode.Uri.file(mainJava);
+        let doc : TextDocument = await vscode.workspace.openTextDocument(u);
+        let e : TextEditor = await vscode.window.showTextDocument(doc);
+
+        try {
+            let terminated = false;
+            let r = new Promise((resolve, reject) => {
+                function waitUserApplication(cnt : number, running: boolean, cb : () => void) {
+                    ps.lookup({
+                        command: "^.*[/\\\\]java",
+                        arguments: "pkg.Main"
+                    }, (err, list ) => {
+                        let success : boolean = (list && list.length > 0) == running;
+                        if (success) {
+                            cb();
+                        } else {
+                            if (cnt == 0) {
+                                reject("Timeout waiting for user application");
+                                return;
+                            }
+                            setTimeout(() => waitUserApplication(cnt - 1, running, cb), 100);
```

Comment:
Expecting the application gets started in 500ms? We'll see how well that behaves on the CI servers.